src/reconcile/rec_write.c

```diff
@@ -1843,8 +1843,10 @@ __rec_split_write(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_CHUNK *chunk
             return (__wt_set_return(session, EBUSY));
 
         /* If we need to restore the page to memory, copy the disk image. */
-        if (r->cache_write_restore)
+        if (r->cache_write_restore) {
+            multi->supd_restore = true;
```

Comment:
I know this is already merged, but I've still got some questions:
Does this correspond to something other than supd_entries being non-zero?
Are there ever cases where we need to initialize or re-initialize supd_restore? It's a bit weird that it never gets set to anything other than true (I can believe that's OK, but wanted to ask).
Given that we didn't end up going with using a NULL disk pointer, but rather a boolean flag. Can you add an assertion in that there aren't ever saved updates associated with multi structures that have a NULL disk image?

Reply:
No, they are different. Here we haven't moved anything to the history store yet. So supd_entries can be non zero but r->cache_write_restore cam be false.

Lacking initialization is actually a problem. I'll fix that in WT-5527.

We can't add the assertion here as I said we haven't moved the saved updates to the history store yet.

Reply:
I can try putting the assert later when we read things back into memory.